internal/transport/grpchttp2/http2bridge_test.go

```diff
@@ -0,0 +1,625 @@
+	}
+	if gotFlag := Flag(gotHdr[4]); gotFlag != wantHdr.Flags {
+		errors = append(errors, fmt.Errorf("Flags: got %#x, want %#x", gotFlag, wantHdr.Flags))
+	}
+	if gotSID := readUint32(gotHdr[5:9]); gotSID != wantHdr.StreamID {
+		errors = append(errors, fmt.Errorf("StreamID: got %d, want %d", gotSID, wantHdr.StreamID))
+	}
+	return errors
+}
+
+// Tests and verifies that the framer correctly reads a Data Frame.
+func (s) TestBridge_ReadFrame_Data(t *testing.T) {
+	c := &testConn{}
+	recvData := "test data"
+	// Writing a Data Frame to the reading buf with recvData as payload.
+	c.rbuf = append(c.rbuf, 0, 0, byte(len(recvData)), byte(FrameTypeData), byte(FlagDataEndStream))
+	c.rbuf = appendUint32(c.rbuf, 1)
+	c.rbuf = append(c.rbuf, []byte(recvData)...)
+
+	f := NewFramerBridge(c, c, 0, nil)
+	fr, err := f.ReadFrame()
+	if err != nil {
+		t.Fatalf("ReadFrame(): %v", err)
+	}
+
+	wantHdr := &FrameHeader{
+		Size:     uint32(len(recvData)),
+		Type:     FrameTypeData,
+		Flags:    FlagDataEndStream,
+		StreamID: 1,
+	}
+
+	if diff := cmp.Diff(fr.Header(), wantHdr); diff != "" {
+		t.Errorf("ReadFrame(): %s", diff)
+	}
+
+	df := fr.(*DataFrame)
+	if string(df.Data) != recvData {
+		t.Errorf("ReadFrame(): Data: got %q, want %q", string(df.Data), recvData)
+	}
+	df.Free()
+}
+
+// Tests and verifies that the framer correctly reads a RSTStream Frame.
+func (s) TestBridge_ReadFrame_RSTStream(t *testing.T) {
+	c := &testConn{}
+	// Writing a RSTStream Frame to the reading buf with ErrCodeProtocol as
+	// payload.
+	c.rbuf = append(c.rbuf, 0, 0, 4, byte(FrameTypeRSTStream), 0)
+	c.rbuf = appendUint32(c.rbuf, 1)
+	c.rbuf = appendUint32(c.rbuf, uint32(ErrCodeProtocol))
+
+	f := NewFramerBridge(c, c, 0, nil)
+	fr, err := f.ReadFrame()
+	if err != nil {
+		t.Fatalf("ReadFrame(): %v", err)
+	}
+
+	wantHdr := &FrameHeader{
+		Size:     4,
+		Type:     FrameTypeRSTStream,
+		Flags:    0,
+		StreamID: 1,
+	}
+	if diff := cmp.Diff(fr.Header(), wantHdr); diff != "" {
+		t.Errorf("ReadFrame(): %s", diff)
+	}
+	rf := fr.(*RSTStreamFrame)
+	if rf.Code != ErrCodeProtocol {
+		t.Errorf("ReadFrame(): Code: got %#x, want %#x", rf.Code, ErrCodeProtocol)
+	}
+}
+
+// Tests and verifies that the framer correctly reads a Settings Frame.
+func (s) TestBridge_ReadFrame_Settings(t *testing.T) {
+	c := &testConn{}
+	s := Setting{ID: SettingsHeaderTableSize, Value: 200}
+	// Writing a Settings Frame to the reading buf with s as payload.
+	c.rbuf = append(c.rbuf, 0, 0, 6, byte(FrameTypeSettings), 0)
+	c.rbuf = appendUint32(c.rbuf, 0)
+	c.rbuf = append(c.rbuf, byte(s.ID>>8), byte(s.ID))
+	c.rbuf = appendUint32(c.rbuf, s.Value)
+
+	f := NewFramerBridge(c, c, 0, nil)
+	fr, err := f.ReadFrame()
+	if err != nil {
+		t.Fatalf("ReadFrame(): %v", err)
+	}
+
+	wantHdr := &FrameHeader{
+		Size:     6,
+		Type:     FrameTypeSettings,
+		Flags:    0,
+		StreamID: 0,
+	}
+	if diff := cmp.Diff(fr.Header(), wantHdr); diff != "" {
+		t.Errorf("ReadFrame(): %s", diff)
```

Comment:
optional: cmp.Diff 's output has this format `(-got, +want)` 

```suggestion
		t.Errorf("ReadFrame() (-got, +want): %s", diff)
```